data/transactions/logic/box.go

```diff
@@ -165,9 +169,22 @@ func MakeBoxKey(appIdx basics.AppIndex, name string) string {
 	   The "bx:" prefix is so that the kvstore might be usable for things
 	   besides boxes.
 	*/
-	key := make([]byte, 3 /* bx: */ +8 /* appIdx, big-endian */ +len(name))
-	copy(key, "bx:")
-	binary.BigEndian.PutUint64(key[3:], uint64(appIdx))
-	copy(key[11:], name)
+	key := make([]byte, boxNameIndex+len(name))
+	copy(key, boxPrefix)
+	binary.BigEndian.PutUint64(key[boxPrefixLength:], uint64(appIdx))
+	copy(key[boxNameIndex:], name)
 	return string(key)
 }
+
+// SplitBoxKey extracts an appid and box name from a string that was created by MakeBoxKey()
+func SplitBoxKey(key string) (basics.AppIndex, string, error) {
+	if len(key) < boxNameIndex {
+		return 0, "", fmt.Errorf("SplitBoxKey() cannot extract AppIndex as key (%s) too short (length=%d)", key, len(key))
+	}
+	if key[:boxPrefixLength] != boxPrefix {
+		return 0, "", fmt.Errorf("SplitBoxKey() illegal app box prefix in key (%s). Expected prefix '%s'", key, boxPrefix)
+	}
+	keyBytes := []byte(key)
+	app := basics.AppIndex(binary.BigEndian.Uint64(keyBytes[boxPrefixLength:boxNameIndex]))
+	return app, key[boxNameIndex:], nil
```

Comment:
what if len(key)!=boxNameIndex?

Reply:
The case `len(key) < boxNameIndex` is handled above which leaves only the case `len(key) > boxNameIndex` to address your question. That case is perfectly fine, since `key[boxNameIndex:]` stores the box name